File: 102. Binary Tree Level Order Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> res;
        if (root == NULL)
            return res;        
        vector<int> level_nodes;
        queue<TreeNode*> nodes;
        TreeNode* p;        
        int level_count = 1;
        nodes.push(root);        
        while(!nodes.empty()){    
            p = nodes.front();
            nodes.pop();                      
            level_nodes.push_back(p->val);            
            level_count--;
            if (p->left != NULL)
                nodes.push(p->left);
            if (p->right != NULL)
                nodes.push(p->right);                        
            if (level_count == 0){
                level_count = nodes.size();                
                res.push_back(level_nodes);
                level_nodes.clear();
            }
        }
        return res;        
    }
};

File: Python/102. Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

import Queue as queue


class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        if not root:
            return []
        nq = queue.Queue()
        level = queue.Queue()
        nq.put(root)
        level.put(1)
        ans = []
        while not nq.empty():
            node = nq.get()
            curr = level.get()
            if len(ans) < curr:
                ans.append(list())
            ans[-1].append(node.val)
            if node.left != None:
                nq.put(node.left)
                level.put(curr + 1)
            if node.right != None:
                nq.put(node.right)
                level.put(curr + 1)
        return ans
